core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -1962,6 +1962,21 @@ private[spark] object Utils extends Logging {
     path
   }
 
+  /**
+   * Updates Spark config with properties from a set of Properties.
+   * Provided properties have the highest priority.
+   */
+  def updateSparkConfigFromProperties(
+      conf: SparkConf,
+      properties: Map[String, String]) : Unit = {
+    properties.filter { case (k, v) =>
+      k.startsWith("spark.")
+    }.foreach { case (k, v) =>
+      conf.set(k, v)
+      sys.props.getOrElseUpdate(k, v)
```

Comment:
Why is this method updating system properties? The name nor the scaladoc mention that fact, and this is a pretty important thing the method does. Not only that but the behavior is different from what it does to the SparkConf (set if not yet set, instead of always set).